### PR TITLE
Doc/response.py

### PR DESCRIPTION
**Pull request by pseyfert**
adding docstrings

### CODE REVIEW

File: libfulltext/response.py [hunk 1]
```diff
@@ -4,17 +4,36 @@
...
+    """Returns a function that checks a save_response call via expected SHA1 hash and path
+
+    Args:
+        expected_sha1: expected SHA1 hash of the downloaded file
+        expected_path: expected basename of the downloaded file
+
+    Returns:
+        A function that takes a stream (from a response) and a string
+        (basename where the stream should usually be saved). To match the
+        interface of save_stream
```
Comment (mfherbst) [inline libfulltext/response.py]: The last sentence is kind of a half-sentence ;)

File: libfulltext/response.py [hunk 2]
```diff
@@ -26,12 +45,22 @@ def save_to_file(response, filename):
...
+    """
     response.raise_for_status()
 
     if 'Content-Type' not in response.headers or \
             response.headers['Content-Type'] != expected_content_type:
-        raise requests.exceptions.InvalidHeader('Content-Type is not application/pdf, '
-                                                'possibly because you do not have access to '
-                                                'this document.')
+        raise requests.exceptions.InvalidHeader('Content-Type is not ' +
+                                                expected_content_type + ', '
```
Comment (mfherbst) [inline libfulltext/response.py:64]: No need for either `+`. Python concatenates strings inside brackets.

**Review by mfherbst — DISMISSED** [review]
A few nits only.

File: libfulltext/response.py (hunk 2 above)
Comment (pseyfert) [reply libfulltext/response.py:64]: Hm, nose and flake complain about `E999 SyntaxError: invalid syntax` upon removal of the second `+`

Comment (mfherbst) [reply libfulltext/response.py:64]: Oh, I'm sorry I misread `expected_content_type` by `'expected_content_type'` for some reason.

Comment (mfherbst) [reply libfulltext/response.py:64]: Should be all good in the way it was.

**Review by mfherbst — APPROVED** [review]

File: libfulltext/response.py (hunk 1 above)
Comment (pseyfert) [reply libfulltext/response.py]: changed to "This function matches the interface of save_stream."

**Pull request merged**